CS3219-AY2223S1/cs3219-project-ay2223s1-g12
# Implement Account Deletion

Fixes #20 

Summary:
* Add feature that allows user to delete their account
* Through the API - `DELETE /api/user/`, requiring `username` to be provided in the json body
* User will be authenticated and authorised first before account deletion is allowed
* User is automatically logged out after account deletion
* Block deletion of other user's account, if you are not logged in as that user

To be done:
* Perform verification of user existence before deleting the user (pending approval of PR for #19) [DONE]
* Blacklist JWT token (pending implementation of login/logout) [DONE]

To Test:
1. `npm i` and `npm run dev` to start the service.
2. Login via `POST /api/user/login` with your `username` and `password` in the json body
3. Proceed to delete account at `DELETE /api/user/` with your `username` in the json body. You should receive a success message that your account was deleted.
4. If you attempt to call `DELETE /api/user/` before logging in, the request will fail.
